### DISCUSSION TITLE
NetBox housekeeping cron job logs in email

### DISCUSSION THREAD

**Discussion by ShreyasSurendra** [netbox-community/netbox#9162] [Q&A] [answered]
I'm trying to prevent the output of the housekeeping cron job to be sent via email. 

Is there a setting that would allow NetBox to handle this? If not, I think one of the ways to prevent email is to redirect the output to a log file, but would like to get some advice on the best/standard approach to handle this.

**Output of cron job**:
/etc/cron.daily/netbox-housekeeping:

[*] Clearing expired authentication sessions
	Sessions cleared.
[*] Checking for expired changelog records
	Deleting 14453 expired records... Done.
[*] Checking for expired jobresult records
	No expired records found.
[*] Checking for latest release
	Skipping: RELEASE_CHECK_URL not set
Finished.

**Comment by mtinberg** [comment, marked as answer]
Cron configuration is not really within the scope of Netbox itself, it's a system service that many things use (see man 5 crontab for the full documentation), but one technique I use when I don't want to get the default reporting emails with the script output is to pipe to /usr/bin/logger to convert the output to syslog messages eg.

```
#Ansible: Netbox daily housekeeping
@daily netbox /opt/netbox/current/venv-py3/bin/python /opt/netbox/current/netbox/manage.py housekeeping | /usr/bin/logger -p daemon.info -t netbox-housekeeping 2>&1
```

that keeps my inbox smaller and if I do need to look a the output of a historical run of the script, I can get it from the logs with the normal retention rules that apply there.

**Reply by candlerb** [reply]
Another option is to use [cronic](https://manpages.ubuntu.com/manpages/focal/man1/cronic.1.html) which captures the stdout, and only forwards it if the script's exit code is non-zero.

**Reply by larsen0815** [reply]
Another option is to use less verbose output (I assume errors would still be printed out).
`sed -i 's:\(/opt/netbox/netbox/manage.py housekeeping\):\1 -v 0:' /opt/netbox/contrib/netbox-housekeeping.sh`
